lib/resources/user.rb

```diff
@@ -53,40 +53,42 @@ def initialize(user)
 
     # select package manager
     @user_provider = nil
-    case inspec.os[:family]
-    when 'ubuntu', 'debian', 'redhat', 'fedora', 'centos', 'arch', 'opensuse', 'wrlinux'
+    os = inspec.os
+    if os.linux?
       @user_provider = LinuxUser.new(inspec)
-    when 'windows'
+    elsif os.windows?
       @user_provider = WindowsUser.new(inspec)
-    when 'darwin'
+    elsif ['darwin'].include?(os[:family])
       @user_provider = DarwinUser.new(inspec)
-    when 'freebsd'
+    elsif ['freebsd'].include?(os[:family])
       @user_provider = FreeBSDUser.new(inspec)
-    when 'aix'
+    elsif ['aix'].include?(os[:family])
       @user_provider = AixUser.new(inspec)
+    elsif os.solaris?
+      @user_provider = SolarisUser.new(inspec)
     else
       return skip_resource 'The `user` resource is not supported on your OS yet.'
     end
   end
 
   def exists?
-    !identiy.nil? && !identiy[:user].nil?
+    !identity.nil? && !identity[:user].nil?
```

Comment:
that last `identity[:user]` would not return a boolean; at least keep `.nil?` there imho